parbase/FairRuntimeDb.cxx

```diff
@@ -461,8 +461,12 @@ Bool_t FairRuntimeDb::writeContainer(FairParSet* cont, FairRtdbRun* run, FairRtd
       }
       break;// End of Ascii IO
     default: // Unknown IO
-      Error("writeContainer()","Unknown output file type.");
-      break;
+      if (cont->hasChanged()) {
+        cv = findOutputVersion(cont);
+        if(cv == 0) {
+          cont->write(output);
+        }
+      }
```

Comment:
* I think all those case statements should be moved to `FairParSet::write(FairParIo*)` overrides in the corresponding child classes and your proposed addition should go into the base class.
   * Then replace the whole `switch(ioType)` statement with just something like `cont->write(output)`
* Then we should get rid of the `FairRuntimeDb::ioType` member (`FairRuntimeDb::isRootFileOutput` can stay for the scope of this PR)
* Please also remove the `USE_DB_METHOD` macro from this file entirely (including comments) and also all code which is no longer enabled, because it is anyways hard-coded to the value `1`

Reply:
Please make 3 separate commits, one for each of the above main bullet points.

Reply:
I do not think that this code should be moved to the inheritors of `FairParSet` but to inheritors `FairDetParIo`, like `FairDetParRootFileIo`. This does not seem to be trivial though. I can attempt to, but I am not quite sure the implementation will be proper

Reply:
Also dev is incompatible with my current fairsoft version, will take time to do the compilation.
Can you please go ahead and implement the proposed changes?

Reply:
Also dev is incompatible with my fairsoft version and compiling the compatible version will take its long time.
Can you please go ahead and implement the suggested changes? Thanks.

Reply:
> I do not think that this code should be moved to the inheritors of `FairParSet` but to inheritors `FairDetParIo`, like `FairDetParRootFileIo`.

You are absolutely right, I confused these two `write`s:
* `Int_t FairParSet::write(FairParIo*)`
* `Int_t FairDetParIo::write(FairParSet*)`

Whose purpose I do not yet understand tbh. Will have to have another deeper look at this.

> This does not seem to be trivial though. I can attempt to, but I am not quite sure the implementation will be proper
> Can you please go ahead and implement the suggested changes? Thanks.

Sure, but unfortunately not immediately, I will try to squeeze it in next week or so. Can you comment on what exactly might be non-trivial (so I do not miss it)?

Reply:
I do, similarly to you, struggle to understand all the logic and purpose of these `ParIo`s. Writing logic is scattered across many classes and not very transparent.